pkg/apis/apps/v1alpha1/servicebinding_types.go

```diff
@@ -18,96 +18,92 @@ type ServiceBindingRequestSpec struct {
 	// +optional
 	EnvVarPrefix string `json:"envVarPrefix,omitempty"`
 
-	// Custom env variables
+	// DataMapping is used to map CR/CSV/CRD attributes to Custom env variables
 	// +optional
-	CustomEnvVar []corev1.EnvVar `json:"customEnvVar,omitempty"`
-
-	// BackingServiceSelector is used to identify the backing service operator.
-	// Deprecation Notice:
-	// In the upcoming release, this field would be depcreated. It would be mandatory
-	// to set "backingServiceSelectors".
-	// +optional
-	BackingServiceSelector *BackingServiceSelector `json:"backingServiceSelector,omitempty"`
+	DataMapping []corev1.EnvVar `json:"dataMapping,omitempty"`
 
 	// BackingServiceSelectors is used to identify multiple backing services.
-	// This would be made a required field after 'BackingServiceSelector'
+	// This would be made a required field after 'services'
```

Comment:
This comment doesn't make sense.

The original comment was about a single `BackingServiceSelector` which is completely removed here in favour of `Services`.

Reply:
FYI, `Services` should be optional field.

Reply:
Ah yes, this comment can go.

Reply:
Done, https://github.com/redhat-developer/service-binding-operator/pull/495/commits/b4dc1abda582e2fe27d93527042738efdc764031 I'll improve the doc further.

Reply:
@DhritiShikhar  I don't think services should be optional. We at least need one service(be it any {Kind}) for binding